### PR TITLE
Makefile: add `bench` convenience target for fishtesters

### PR DESCRIPTION
**Pull request by dubslow**
It's just an alias for `make build && ./stockfish bench | grep -v crap`, which is the usual way to write a fishtest. It's small but it saves me the pain of having to remember the command or find it in my shell history... it makes writing fishtests that little bit easier.

### CODE REVIEW

**Comment by MinetaS** [comment]
I see there are no reasons to keep displaying positions if you run bench only for fishtest, hence `tail -n 4` might be slightly better...

**Comment by XInTheDark** [comment]
What if an error occurred during build? Then it would output the bench for the previously built binary right?

**Comment by XInTheDark** [comment]
> I see there are no reasons to keep displaying positions if you run bench only for fishtest, hence `tail -n 4` might be slightly better...

Idk, it might be useful in some cases? Same reason for why it's displayed in `bench` I guess...?

**Comment by vondele** [comment]
`make profile-build` also shows you the bench?

**Comment by XInTheDark** [comment]
> `make profile-build` also shows you the bench?

For me, the `make profile-build` process is much slower than `make build` then `./stockfish bench`.

**Comment by dubslow** [comment]
> I see there are no reasons to keep displaying positions if you run bench only for fishtest, hence tail -n 4 might be slightly better...

Being not very proficient with Stockfish, I frequently write bad patches, and in such cases seeing the positions scroll by confirms to me that the bench is in fact working properly. A somewhat different usecase than `profile-build'`s `tail`.

I do with I could get the blank lines cut out tho. When I tested it locally, running `grep -Ev '^$|^info|^best'` does indeed cut the blank lines as intended, but for whatever reason it doesn't work when run from the Makefile. Dunno why.

> What if an error occurred during build? Then it would output the bench for the previously built binary right?

If an error occurs in the build, then the `build` dependency is not met and the rest of it doesn't run. 

> `make profile-build` also shows you the bench?

True, but that takes 10x longer in typical cases. Usually we only modify a file or two for fishtest, so `build` is pretty fast, whereas `profile-build` necessarily rebuilds *everything* in the codebase, *twice*. I used to use this as my bench producer when I started, but it's simply far too long to rattle out fishtest patches.

**Comment by silversolver1** [comment]
Maybe this is just a brain fart on my part, but I have been using profile-build in my build script this whole time...

The older wiki also had this in place under the 64-bit windows script: https://github.com/glinscott/fishtest/wiki/Building-Stockfish-on-Windows/6969e5b6052f1956fc8c3e28903dba8254fa1be6#building-stockfish-with-gcc

It's no wonder the cpu usage was constantly so high and the compilation always took so long...

In retrospect this is obvious but...

Maybe we can make this more explicitly clear in the new wiki?: 
https://github.com/official-stockfish/Stockfish/wiki/Compiling-from-source#lower-compilation-time
https://github.com/official-stockfish/Stockfish/wiki/Compiling-from-source#general

Specifically, we should make it very clear that build is much a faster process and much easier on the CPU, and that just for purposes of testing, when you don't want actual performance in terms of gameplay or ELO (ie just for a fast build/compilation time), normal build should be preferred over profile-build

Quite simply this wasn't obvious and once the script worked, I just kept using it

TLDR-make it obvious that build may be preferred over profile-build in some use-cases (speed of compilation)

**Comment by vondele** [comment]
for fast compilation and development use `make -j ARCH=x86-64 build optimize=no debug=yes` this takes about 3.0s on my system, and enables the asserts that are helpful for people developing code. Other than for developers, I see no reason to use the `build` target, and I would expect devs to have read `make help`.

However, despite doing probably more SF compiles than most people, I'm almost always using `make -j ARCH=x86-64-avx2 profile-build` at about 20s. This is still a very workable compilation time and gives production quality compiler.

**Comment by dubslow** [comment]
@silversolver1 if you're building to use as a chess engine, then `profile-build` is the way to go. if you're building as a developer, such as to make fishtests, then `build` is much faster.

@vondele indeed `build` and in turn `bench` are both aimed at developers. 20s vs 3s makes a lot of difference in rattling out fishtests. however silversolver1 makes a good point that perhaps what profile-build does is underdocumented in make help

**Comment by silversolver1** [comment]
Thanks for the tips @dubslow @vondele

**Comment by dubslow** [comment]
Hm maybe instead rename `bench` to `debug` and have this target add the `optimize=no debug=yes` by default? I've never even considered using those options since I've just used the first thing that worked.

**Comment by vondele** [comment]
so, I'll close this one, I don't see a real advantage of adding it, it is just one specific workflow.

**Comment by dubslow** [comment]
"fishtesters" is i would argue a very common workflow
